src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -0,0 +1,445 @@
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+template <typename RequestT, typename FuncT>
+void Instance::HandleCommand(HandlerContext & handlerContext, FuncT func)
+{
+    if (!handlerContext.mCommandHandled && (handlerContext.mRequestPath.mCommandId == RequestT::GetCommandId()))
+    {
+        RequestT requestPayload;
+
+        // If the command matches what the caller is looking for, let's mark this as being handled
+        // even if errors happen after this. This ensures that we don't execute any fall-back strategies
+        // to handle this command since at this point, the caller is taking responsibility for handling
+        // the command in its entirety, warts and all.
+        //
+        handlerContext.SetCommandHandled();
+
+        if (DataModel::Decode(handlerContext.mPayload, requestPayload) != CHIP_NO_ERROR)
+        {
+            handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath,
+                                                     Protocols::InteractionModel::Status::InvalidCommand);
+            return;
+        }
+
+        func(handlerContext, requestPayload);
+    }
+}
+
+void Instance::handleChangeToMode(HandlerContext & ctx, const Commands::ChangeToMode::DecodableType & commandData)
+{
+    uint8_t newMode = commandData.newMode;
+
+    Commands::ChangeToModeResponse::Type response;
+
+    if (!IsSupportedMode(newMode))
+    {
+        ChipLogError(Zcl, "ModeBase: Failed to find the option with mode %u", newMode);
+        response.status = static_cast<uint8_t>(StatusCode::kUnsupportedMode);
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        return;
+    }
+
+    HandleChangeToMode(newMode, response);
+
+    if (response.status == static_cast<uint8_t>(StatusCode::kSuccess))
+    {
+        UpdateCurrentMode(newMode);
+        ChipLogProgress(Zcl, "ModeBase: handleChangeToMode changed to mode %u", newMode);
+    }
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+// This function is called by the interaction model engine when a command destined for this instance is received.
+void Instance::InvokeCommand(HandlerContext & handlerContext)
+{
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case ModeBase::Commands::ChangeToMode::Id:
+        ChipLogDetail(Zcl, "ModeBase: Entering handling ChangeToModeWithStatus");
+
+        HandleCommand<Commands::ChangeToMode::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { handleChangeToMode(ctx, commandData); });
+    }
+}
+
+// List the commands supported by this instance.
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster,
+                                               CommandHandlerInterface::CommandIdCallback callback, void * context)
+{
+    callback(ModeBase::Commands::ChangeToMode::Id, context);
+    callback(ModeBase::Commands::ChangeToModeResponse::Id, context);
```

Comment:
We do not access the response, only the ChangeToMode. Remove this line.

Reply:
To be more precise:

```
=== AcceptedCommandList Attribute
This attribute is a list of client generated commands which are supported by this cluster server instance.

Each instance of a cluster SHALL support this attribute.

This attribute SHALL be a list of the command IDs for client generated commands that are supported and processed by the server.

For each client request command in this list that mandates a response from the server,
the response command SHALL be indicated in the GeneratedCommandList attribute.

=== GeneratedCommandList Attribute
This attribute is a list of server generated commands.
A server generated command is a server to client command.

Each instance of a cluster SHALL support this attribute.

This attribute SHALL be a list of the command IDs for server generated commands.

For each command in this list that is a response to a client command request,
the request command SHALL be indicated in the AcceptedCommandList attribute.
```

^^^ The response goes in GeneratedCommandList, not AcceptedCommandList.

Reply:
Done.